### PR TITLE
#25187 Add a new URL Parameter Goal for experiment

### LINKED ISSUE #25187: [UI] New Goal `URL Parameter` to experiment
### Parent Issue

#22455

### User Story

As a user,  
I want to create a new goal for Experiments called 'URL Parameter' so that I can detect a specific URL parameter submitted on any page.

### Acceptance Criteria
- Add the New Goal to the existing sidebar of `select goal`
- Name of the new goal: `URL Parameter`
-  Description of the new goal: `Detect a specific URL parameter submitted on any page`
- should include a required input field to enter the name of the query parameter.
- should have a required dropdown menu to select the operator, with options: `Exists`, `Contain`, and `Equal`
-  All the titles, descriptions, placeholders, and labels need to be translated

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Mocks
Screenshot 2023-06-08 at 12 10 47 PM

Screenshot 2023-06-08 at 12 07 27 PM

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* New goal `URL Parameter`

### Checklist
- [x] Tests
- [x] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Screenshots

Screenshot 2023-06-13 at 4 38 43 PM

Goal Selected
Screenshot 2023-06-13 at 4 38 48 PM

### <samp>🤖 Generated by Copilot at b0f2b17</samp>

This pull request adds a new goal type of URL parameter for the dot-experiments module, and updates the UI components, models, constants, validators, and translations to support the new goal type. It also improves the performance, readability, and error handling of the existing components and code.

## Related Issue
Fixes #25187 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at b0f2b17</samp>

*  Add a new goal type for URL parameter and its configuration form ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-68dabaa4a62686c040e5519c0bafc3f729a426dd92755d27c2b3078078912a93R38-R47), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-087507badb1f1a9ccc4e7907c1f60aa95de989dfc1838492bf171d419c05ee33R1-R75), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-2f94b01385305626773dea035125bed468e84ef7d7fd2248899e0f44373478deR1-R38), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-5ab1dfc0b7f06b2da23665bd1cbe5a292a5a19036c67dc4478a3158ff53784e9R1-R16), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-42a06796f5e2d75bfc763218a926ab985956719551af92772e07edfc0493c428R1-R40), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-f1a1a6a6f21372c106dd50983b66d511523dd176656f6d368fe830a87a26b09cL64-R116), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-f1a1a6a6f21372c106dd50983b66d511523dd176656f6d368fe830a87a26b09cR149-R152), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-253959797c88a1b66365ca0759703481a484a03c59dc027e2dfc6accc6cf2b77L151-R170), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-253959797c88a1b66365ca0759703481a484a03c59dc027e2dfc6accc6cf2b77R235-R236), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94L5225-R5236), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94L5236-R5247), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94R5313))
*  Extend the base component for the goal configuration components and refactor the reach page component to use it ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-98e2f4f3ae36ceb89218245d175beba226d830016c08be6aa000004ac4d8de49L25-R25), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-98e2f4f3ae36ceb89218245d175beba226d830016c08be6aa000004ac4d8de49L44-R44), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-98e2f4f3ae36ceb89218245d175beba226d830016c08be6aa000004ac4d8de49L55), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-98e2f4f3ae36ceb89218245d175beba226d830016c08be6aa000004ac4d8de49L66-R65), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-756b31aa08fe2fbd305c8d4a7c5608a2d1789f51a08b469ff5306c4d6ef85da5L2-R17), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-756b31aa08fe2fbd305c8d4a7c5608a2d1789f51a08b469ff5306c4d6ef85da5L22-R36), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-756b31aa08fe2fbd305c8d4a7c5608a2d1789f51a08b469ff5306c4d6ef85da5L35-R48), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-459a6168db42fbf9138c286c7398dfa339c41a7dc683e9ba51fe9fff6e4ec897R1-R34), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-890f9e278647b1f7fe0260aef354548d1a611624dbbd55456f3b3787da60c1b0R1-R22))
*  Add error handling logic to the saveGoal method in the store ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-7a90616a0732d0f6b7355c5fc0be32eb154c777900d8d29d700960e9dbaeb5a0L535-R540))
*  Modify the valueChanges subscription of the primary type control to remove the validations from the conditions form array when the selected type is bounce rate ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-e1586d62dab99af430d2dfa25a918712538703b64f827f8ee459b2dd8289a13aL148-R161))
*  Modify the defineDefaultName method to use a defaultNameValuesByType object and handle the URL parameter type ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-e1586d62dab99af430d2dfa25a918712538703b64f827f8ee459b2dd8289a13aL189-R252))
*  Add a condition to only render the content template of the option item if the item is expanded ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-c0b6e2a33f9988def9c0549b0e1b03144ae3a45ab69ef11b2de735ee4592274fL18-R20))
*  Modify the style of the detail and title paragraphs of the option item to improve the contrast and readability ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-af4267fc43859821b41f8089ce3df9ce945560bd66333f83a516ef36c741676fL25-R25), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-af4267fc43859821b41f8089ce3df9ce945560bd66333f83a516ef36c741676fL35-R35))
*  Add a new validator method for the query param name and refactor the alphaNumeric validator method ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-f2a9d3e64396c1090a1c089d2b49d9c9e145e80e4a03c3d0311eb0be39f6c343L3-R32))
*  Remove the unused imports and dependency injections from the goal select component ([link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-e1586d62dab99af430d2dfa25a918712538703b64f827f8ee459b2dd8289a13aL4-R12), [link](https://github.com/dotCMS/core/pull/25226/files?diff=unified&w=0#diff-e1586d62dab99af430d2dfa25a918712538703b64f827f8ee459b2dd8289a13aL80-R102))

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        color: $color-palette-gray-500;
+    }
+
+    h3 {
+        font-size: $font-size-md;
+        margin: $spacing-0;
+        margin-bottom: $spacing-1;
+        font-weight: $font-weight-bold;
+    }
+}
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss:30]: We already have a `.field` class that style the labels... we need to keep the forms labels consistent.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+.condition {
+    gap: $spacing-3;
+
+    label {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss:35]: Same here.

File: core-web/package-lock.json
Comment (fmontes) [inline core-web/package-lock.json:1]: Why?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss:30]: Let me take a look.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss (hunk 2 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-goal-configuration-url-parameter-component/dot-experiments-goal-configuration-url-parameter-component.component.scss:35]: same here.

File: core-web/package-lock.json
Comment (oidacra) [reply core-web/package-lock.json:1]: Removed from the changes.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts [hunk 3]
```diff
@@ -128,7 +128,7 @@ describe('DotExperimentsConfigurationGoalSelectComponent', () => {
     it('should have rendered BOUCE_RATE and REACH_PAGE options items', () => {
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts:128]: RequiredConditionsControlsByGoalTypeRules

**Review by hmoreras — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/components/dot-experiments-option-content-base-component/dot-experiments-option-content-base.component.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { FormArray, FormGroup, FormGroupDirective } from '@angular/forms';
+
+@Component({
+    standalone: true,
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotExperimentsOptionContentBaseComponent {
+    parametersList;
+    operatorsList;
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/components/dot-experiments-option-content-base-component/dot-experiments-option-content-base.component.ts:11]: What is this for?

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/components/dot-experiments-option-content-base-component/dot-experiments-option-content-base.component.ts (hunk 4 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/components/dot-experiments-option-content-base-component/dot-experiments-option-content-base.component.ts:11]: properties for the child components.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts (hunk 3 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts]: Added URL_PARAMETERS

**Pull request merged**